lua/usip_player.lua

```diff
@@ -2380,38 +2416,49 @@ if fIsSecure and not tArgs.fCommandVerifyHelperSignaturesSelected then
     else
         -- verify the signature of the used HTBL files
         -- make a list of necessary files
-        local tblHtblFilePaths = {}
+        local tblHtblFileData = {}
+        local tPathList = {}
         local fDoVerify = false
         if (tArgs.fVerifySigEnable or not tArgs.fVerifyContentDisabled) then
             fDoVerify = true
-            table.insert(tblHtblFilePaths, strReadSipPath)
+            strFileData, _ = tFlasherHelper.loadBin(strReadSipPath)
+            table.insert(tblHtblFileData, strFileData)
+            table.insert( tPathList, strReadSipPath)
         end
         if tArgs.strBootswitchParams then
             fDoVerify = true
             if tArgs.strBootswitchParams == "JTAG" then
-                table.insert( tblHtblFilePaths, strExecReturnFilePath )
+                strFileData, _ = tFlasherHelper.loadBin(strExecReturnPath)
+                table.insert( tblHtblFileData, strFileData)
+                table.insert( tPathList, strExecReturnPath)
             else
-                table.insert( tblHtblFilePaths, strBootswitchFilePath )
+                strFileData, _ = tFlasherHelper.loadBin(strBootswitchFilePath)
+                table.insert( tblHtblFileData, strFileData)
+                table.insert( tPathList, strBootswitchFilePath)
             end
         end
 
         -- TODO why not verify set_kek.bin even if no bootswitch was selected?
```

Comment:
I think the list of files to be checked is not up to date, I'll have a look at this.